### DISCUSSION TITLE
Modify response before returning?

### DISCUSSION THREAD

**Discussion by michilehr** [thecodingmachine/graphqlite#524] [Q&A] [answered]
Hi,

is it possible to modify a response before sending?

Given I have a Configuration with a name that differs from language to language (de and en). 

```
/**
 * @Type()
 */
class Configuration
{
    private string $name;
    
    public function getName(): ?string
    {
        return $this->name;
    }
}
```

Now I want to query for the name...

```
query {
  configurations {
	name
  }
} 
```

... and the response should look like this:

```
{
  "data": {
    "configurations": {
      "de": {
        "name": "config de"
      },
      "en": {
        "name": "config en"
      }
    }
  }
}
```

Would this even be valid by the GraphQL specification?

Thanks for your help!

**Comment by oojacoboo** [comment, marked as answer]
Hey Michi - of course.  It might help to use an extended output type.  Checkout this doc:

https://graphqlite.thecodingmachine.io/docs/3.0/extend_type

On your output, the structure of the JSON should really follow your GQL.  So...

```gql
query {
  configurations {
        lang {
	    name
        }
  }
}
```

**Reply by michilehr** [reply]
Thank you for your quick response @oojacoboo!
